docs/design/configuration.md

```diff
@@ -0,0 +1,141 @@
+# Kubernetes Component Configuration
+
+## Abstract
+
+A generic ``Configuration`` object to store command-line flags that is used for
```

Comment:
Agreed.

Furthermore, I've already heard of people (ab)using Secrets for this purpose. We either need to generalize Secret or create a non-Secret alternative.